tests/src/test/scala/cats/tests/ComposeTests.scala

```diff
@@ -0,0 +1,58 @@
+package cats
+package tests
+
+import cats.data.{ NonEmptyList, NonEmptyVector, OneAnd }
+import cats.laws.discipline.{ ApplicativeTests, FoldableTests, MonoidalTests, SemigroupKTests, arbitrary, eq }, arbitrary._, eq._
+import org.scalacheck.Arbitrary
+
+class ComposeTests extends CatsSuite {
+  // we have a lot of generated lists of lists in these tests. We have to tell
+  // Scalacheck to calm down a bit so we don't hit memory and test duration
+  // issues.
+  implicit override val generatorDrivenConfig: PropertyCheckConfiguration =
+    PropertyCheckConfig(maxSize = 5, minSuccessful = 20)
+
+  {
+    // Applicative composition
+
+    implicit val applicativeListVector: Applicative[Lambda[A => List[Vector[A]]]] = Applicative[List] compose Applicative[Vector]
+    implicit val iso = MonoidalTests.Isomorphisms.invariant[Lambda[A => List[Vector[A]]]]
+
+    checkAll("Applicative[Lambda[A => List[Vector[A]]]]", ApplicativeTests[Lambda[A => List[Vector[A]]]].applicative[Int, Int, Int])
+  }
+
+  {
+    // Foldable composition
+
+    implicit val foldableListVector: Foldable[Lambda[A => List[Vector[A]]]] = Foldable[List] compose Foldable[Vector]
+
+    checkAll("Foldable[Lambda[A => List[Vector[A]]]]", FoldableTests[Lambda[A => List[Vector[A]]]].foldable[Int, Int])
+  }
+
+  {
+    // Reducible composition
+
+    val nelReducible =
+      new NonEmptyReducible[NonEmptyList, List] {
+        def split[A](fa: NonEmptyList[A]): (A, List[A]) = (fa.head, fa.tail)
+      }
+
+    val nevReducible =
+      new NonEmptyReducible[NonEmptyVector, Vector] {
+        def split[A](fa: NonEmptyVector[A]): (A, Vector[A]) = (fa.head, fa.tail)
+      }
```

Comment:
I was surprised to see that these aren't already provided. I consider it outside of the scope of this PR, but I think we should follow up on this and add these instances in the `OneAnd` companion object.